From 44db4d978c9524c42e8853f0e2363cc9694bc9ae Mon Sep 17 00:00:00 2001
From: Christopher Champagne <cjchampagne@habana.ai>
Date: Mon, 16 Jun 2025 18:23:33 +0300
Subject: [PATCH] [SW-232459] pdr: set next supported record handle

Set the next record handle field to the next record handle in the
list that is supported on the current platform.

Change-Id: I46b37c941b108093f8f6aa448db8988ff1e9c9bf
Signed-off-by: Christopher Champagne <cjchampagne@habana.ai>
---

diff --git a/src/pldm_pmc.c b/src/pldm_pmc.c
index 8569f6a..1898b13 100644
--- a/src/pldm_pmc.c
+++ b/src/pldm_pmc.c
@@ -150,6 +150,18 @@
 	return iter->header->record_handle + iter->pdr_index;
 }
 
+static uint32_t iter_get_next_supported_handle(struct pdr_header_list_iter *iter)
+{
+	uint8_t current_platform = pldm_get_current_platform();
+
+	for (iter_inc(iter); iter_is_valid(iter); iter_inc(iter)) {
+		if (pldm_pmc_unsupported_pdr(iter->meta, current_platform))
+			continue;
+		return iter_get_handle(iter);
+	}
+	return 0;
+}
+
 static bool find_header_in_list(
 	uint32_t handle,
 	const struct pdr_header_list *list,
@@ -163,7 +175,7 @@
 
 		if (next_handle) {
 			struct pdr_header_list_iter next = *iter;
-			*next_handle = iter_inc(&next) ? iter_get_handle(&next) : 0;
+			*next_handle = iter_get_next_supported_handle(&next);
 		}
 		return true;
 	}
